### ISSUE TITLE
CLPZ: Error: syntax error unexpected

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#528]
v2.50.32
```
?- assertz(clpz:monotonic).
   true.
?- use_module(library(clpz)).
   true.
?- 1 #= #X.
Error: syntax error, near '#=', operator expected, user:1 % unexpected
?-
```

**Comment by infradig** [comment]
Dunno, something to do with goal expansion maybe. Putting the assertz after the
use_module is obviously fine.

Oddly Scryer does this...

$ scryer-prolog
?- assertz(clpz:monotonic).
   true.
?- use_module(library(clpz)).
% Warning: overwriting $clause/2 because the clauses are discontiguous
   true.
?- 1 #= #X.
   X = 1.

So the order seems meaningful at some level.

**Comment by infradig** [comment]
Just noticed you comment on Scryer issues.

**Comment by flexoron** [comment]
Fixed (tested here: v2.52.16) 
```
?- use_module(library(clpz)).
   true.
?- assertz(clpz:monotonic).
   true.
?- use_module(library(clpz)).
   true.
?- 1 #= #X.
   X = 1.
?- 1 #= X.
   throw(error(instantiation_error,instantiation_error(unknown(_5),1))).
?- retract(clpz:monotonic).
   true.
?- 1 #= X.
```
